vnext/Microsoft.ReactNative.Cxx/TurboModuleProvider.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <JSI/JsiAbiApi.h>
+#include <JSI/JsiApiContext.h>
+#include <ReactCommon/TurboModule.h>
+#include <winrt/Microsoft.ReactNative.h>
+#include <winrt/Windows.Foundation.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+// CallInvoker on top of JSDispatcher.
+struct AbiCallInvoker : facebook::react::CallInvoker {
+  AbiCallInvoker(IReactDispatcher const &jsDispatcher) : m_jsDispatcher(jsDispatcher) {}
+
+  void invokeAsync(std::function<void()> &&func) override {
+    m_jsDispatcher.Post([func = std::move(func)]() { func(); });
+  }
+
+  void invokeSync(std::function<void()> &&func) override {
+    if (m_jsDispatcher.HasThreadAccess()) {
+      func();
+    } else {
+      std::mutex mutex;
+      std::condition_variable cv;
+      bool completed{false};
+      auto lock = std::unique_lock{mutex};
+      m_jsDispatcher.Post([&func, &mutex, &cv, &completed]() {
+        func();
+        auto lock = std::unique_lock{mutex};
+        completed = true;
+        cv.notify_all();
+      });
+      cv.wait(lock, [&] { return completed; });
+    }
+    func = nullptr;
```

Comment:
>func = nullptr; [](start = 4, length = 15)

Why is this needed?

Reply:
Added a comment: Since func is passed as an r-value, we want to clean it up in this method as we would move the value.